default.nix

```diff
@@ -1,10 +1,18 @@
 { pkgs ? import <nixpkgs> {}
 , lib ? pkgs.lib
 , poetry ? null
+, poetryLib ? import ./lib.nix { inherit lib; }
```

Comment:
what's the advantage of making the `lib` and `poetryLib` parameters overridable?

Reply:
I guess the same reasons you might want to pass in `poetry`.

for `lib`: Being able to pass in a new version explicitly. I've had more then one case where the current project was on some kind of older nixpkgs revision but I had access to a newer `lib`. Many `x2nix` tools are only developed with unstable or the latest release in mind. Being able to get your stuff working anyway is a great value.

Same for the `poetryLib`. I can imagine being in a situation where I have to customize some kind of behavior of the nix code and not just the derivation. The more control we can expose to users without hurting yourself the better IMO.